### PR TITLE
Ask confirmation for running untrusted resources

### PR DESCRIPTION
**Pull request by jderusse**
Here is a proposal to the issue #32.
The objective, is to avoid the execution of untrusted scripts

* [x] Add tests
* [x] Document it
* [x] Remember trusted signatures
* [x] Trust users

```bash
$ ./bin/melody run ~/test.php  -vvv
Hello Melody!

$ ./bin/melody run 23bb3980daf65154c3d4 
You are running an untrusted resource
  URL: https://gist.github.com/23bb3980daf65154c3d4
  Owner: lyrixx
  Created at: Fri, 17 Oct 2014 14:44:13 +0000
  Last update: Fri, 17 Oct 2014 23:45:48 +0000
  Revision: #2

Do you want to continue [y/N]?y
----------------------------------
Hello greg
----------------------------------

$ ./bin/melody run 23bb3980daf65154c3d4 --trust
----------------------------------
Hello greg
----------------------------------
```

### CODE REVIEW

File: src/SensioLabs/Melody/Console/Command/RunCommand.php [hunk 1]
```diff
@@ -72,11 +82,74 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-        $melody->run($script, $arguments, $configuration, $cliExecutor);
+        while (true) {
+            try {
+                $melody->run($script, $arguments, $configuration, $cliExecutor);
+                return 0;
+            } catch (TrustException $e) {
+                if ($this->confirmTrust($e->getResource(), $input, $output)) {
+                    $configuration->addTrustedSignatures($e->getResource()->getMetadata()->getSignature());
+                } else {
+                    throw new \RuntimeException('Operation aborted by the user.');
```
Comment (stof) [inline src/SensioLabs/Melody/Console/Command/RunCommand.php]: not 0 if it is a failure

**Comment by lyrixx** [comment]
@jderusse Could you rebase ? (and finish this one ;))

**Comment by jderusse** [comment]
rebased and green :)

File: doc/features.md [hunk 2]
```diff
@@ -97,6 +97,52 @@ melody will catch them. To use options, you must prepend your options by
...
+$ melody run 565752f13499a3fa17d9 --trust
+```
+
+User Configuration
+------------------
+
+Melody stores your configuration in a file located in `~/.sensiolabs/melody.yml`.
+This file contains:
+
+* a list of trusted resources signatures (see setion Trust).
```
Comment (romqin) [inline doc/features.md]: section

File: src/SensioLabs/Melody/Console/Command/RunCommand.php [hunk 3]
```diff
@@ -29,6 +33,7 @@ protected function configure()
             ->addArgument('arguments', InputArgument::IS_ARRAY | InputArgument::OPTIONAL, 'Which arguments do you want to pass to the script?')
             ->addOption('no-cache', null, InputOption::VALUE_NONE, 'If set, do not rely on previous cache.')
             ->addOption('prefer-source', null, InputOption::VALUE_NONE, 'Forces installation from package sources when possible, including VCS information.')
+            ->addOption('trust', 't', InputOption::VALUE_NONE, 'Trust all resources.')
```
Comment (romqin) [inline src/SensioLabs/Melody/Console/Command/RunCommand.php]: why in plural ?

**Comment by lyrixx** [comment]
It took time, but here we go, this is in now. Thank you very much @jderusse.

**Pull request closed without merging**